sklearn/ensemble/tests/test_stacking.py

```diff
@@ -0,0 +1,335 @@
+    )
+    reg_drop = StackingRegressor(
+        estimators=estimators, final_estimator=rf, cv=5, random_state=42
+    )
+
+    reg.fit(X_train, y_train)
+    reg_drop.fit(X_train, y_train)
+    assert_allclose(reg.predict(X_test), reg_drop.predict(X_test))
+
+
+@pytest.mark.filterwarnings("ignore:Liblinear failed to converge")
+@pytest.mark.filterwarnings("ignore:The default value of n_estimators")
+@pytest.mark.filterwarnings("ignore:Default solver will be changed to 'lbfgs'")
+@pytest.mark.filterwarnings("ignore:Default multi_class will be changed")
+@pytest.mark.parametrize(
+    "cv",
+    [3, KFold(n_splits=3, shuffle=True, random_state=42),
+     LeaveOneOut()]
+)
+@pytest.mark.parametrize(
+    "final_estimator, predict_params",
+    [(None, {}),
+     (RandomForestRegressor(random_state=42), {}),
+     (DummyRegressor(), {'return_std': True})]
+)
+@pytest.mark.parametrize(
+    "passthrough, X_trans_shape",
+    [(False, 2),
+     (True, 12)]
+)
+def test_stacking_regressor_diabetes(cv, final_estimator, predict_params,
+                                     passthrough, X_trans_shape):
+    X_train, X_test, y_train, y_test = train_test_split(
+        X_diabetes, y_diabetes, random_state=42
+    )
+    estimators = [('lr', LinearRegression()), ('svr', LinearSVR())]
+    reg = StackingRegressor(estimators=estimators,
+                            final_estimator=final_estimator,
+                            cv=cv, passthrough=passthrough, random_state=42)
+    reg.fit(X_train, y_train)
+    reg.predict(X_test, **predict_params)
+    assert reg.score(X_test, y_test) < 0.6
+
+    X_trans = reg.transform(X_test)
+    assert X_trans.shape[1] == X_trans_shape
+
+    reg.set_params(lr=None)
+    reg.fit(X_train, y_train)
+    reg.predict(X_test)
+
+
+class NoWeightRegressor(BaseEstimator, RegressorMixin):
+    def __init__(self):
+        self.reg = DummyRegressor()
+
+    def fit(self, X, y):
+        return self.reg.fit(X, y)
+
+    def predict(self, X):
+        return np.ones(X.shape[0])
+
+
+class NoWeightClassifier(BaseEstimator, ClassifierMixin):
+    def __init__(self):
+        self.clf = DummyClassifier()
+
+    def fit(self, X, y):
+        return self.clf.fit(X, y)
+
+
+@pytest.mark.filterwarnings("ignore:Liblinear failed to converge")
+@pytest.mark.filterwarnings("ignore:The default value of n_estimators")
+@pytest.mark.filterwarnings("ignore:Default solver will be changed to 'lbfgs'")
+@pytest.mark.filterwarnings("ignore:Default multi_class will be changed")
+@pytest.mark.parametrize(
+    "X, y, params, type_err, msg_err",
+    [(X_iris, y_iris,
+     {'estimators': None, 'final_estimator': RandomForestClassifier(),
+      'predict_method': 'auto'},
+      AttributeError, "Invalid 'estimators' attribute,"),
+     (X_iris, y_iris,
+      {'estimators': [('lr', LogisticRegression()), ('svm', LinearSVC())],
+       'final_estimator': RandomForestClassifier(), 'passthrough': 'random'},
```

Comment:
no need to specify final_estimator

(there are other places when you don't need to pass some parameters because the default are enough, I'll stop mentioning them but please remove them)